packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx

```diff
@@ -46,60 +67,242 @@ export class AtomicSearchBox {
-              onClick={() => this.searchBox.clear()}
-              ariaLabel={this.bindings.i18n.t('clear')}
-            >
-              <atomic-icon icon={ClearIcon} class="w-3 h-3"></atomic-icon>
-            </Button>
-          )}
-      </div>
+      this.activeDescendantElement?.previousElementSibling || this.lastValue
     );
   }
 
-  // TODO: adapt for atomic-query-suggestions & atomic-recent-queries
-  private renderSuggestions() {
-    if (!this.searchBoxState.suggestions.length) {
+  private focusNextValue() {
+    if (!this.hasSuggestions) {
       return;
     }
 
-    if (!this.hasFocus) {
+    const query = this.nextOrFirstValue.getAttribute('data-value');
+    !isNullOrUndefined(query) && this.updateQuery(query);
+    this.updateActiveDescendant(this.nextOrFirstValue.id);
+  }
+
+  private focusPreviousValue() {
+    if (!this.hasSuggestions) {
+      return;
+    }
+
+    const query = this.previousOrLastValue.getAttribute('data-value');
+    !isNullOrUndefined(query) && this.updateQuery(query);
+    this.updateActiveDescendant(this.previousOrLastValue.id);
+  }
+
+  private onInput(value: string) {
+    this.searchBox.updateText(value);
+    this.updateActiveDescendant();
+  }
+
+  private onFocus() {
+    this.isExpanded = true;
+    if (!this.searchBoxState.suggestions.length) {
+      this.searchBox.showSuggestions();
+    }
+  }
+
+  private onBlur() {
+    this.isExpanded = false;
+    this.updateActiveDescendant();
+  }
+
+  private onSubmit() {
+    if (this.activeDescendantElement) {
+      this.activeDescendantElement.click();
+      this.updateActiveDescendant();
       return;
     }
 
+    this.searchBox.submit();
+    this.updateActiveDescendant();
+  }
+
+  private onKeyup(e: KeyboardEvent) {
+    switch (e.key) {
+      case 'Enter':
+        this.onSubmit();
+        break;
+      case 'Escape':
+        this.onBlur();
+        break;
+    }
+  }
+
+  private updateQuery(query: string) {
+    this.bindings.engine.dispatch(
+      loadQuerySetActions(this.bindings.engine).updateQuerySetQuery({
+        id: this.id,
+        query,
+      })
+    );
+  }
+
+  private onKeyDown(e: KeyboardEvent) {
+    switch (e.key) {
+      case 'ArrowDown':
+        e.preventDefault();
+        this.focusNextValue();
+        break;
+      case 'ArrowUp':
+        e.preventDefault();
+        this.focusPreviousValue();
+        break;
+    }
+  }
+
+  private renderInput() {
+    return (
+      <input
+        part="input"
+        ref={(el) => (this.inputRef = el as HTMLInputElement)}
+        role="combobox"
+        aria-autocomplete="both"
+        aria-haspopup="true"
+        aria-owns={this.popupId}
+        aria-expanded={`${this.isExpanded}`}
+        aria-activedescendant={this.activeDescendant}
+        autocomplete="off"
+        autocapitalize="off"
+        autocorrect="off"
+        placeholder={this.bindings.i18n.t('search')}
+        aria-label={this.bindings.i18n.t('search-box')}
+        type="text"
```

Comment:
Should it be `type=search` ?

Reply:
no type search add custom stuff we don't want https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search